### ISSUE TITLE
[Bug] isPlainObj fn not compatible for IE11

### ISSUE THREAD

**Issue by binarykitchen** [immutable-js/immutable-js#1745]
This implementation is breaking on IE11

```
  function isPlainObj(value) {
    return (
      value &&
      (typeof value.constructor !== 'function' ||
        value.constructor.name === 'Object')
    );
  }
```

Always returns false.

Refer to https://stackoverflow.com/questions/25140723/constructor-name-is-undefined-in-internet-explorer

**Comment by binarykitchen** [comment]
A workaround would be to polyfill that with

```js
  if ('name' in Object.constructor.prototype === false) {
    Object.defineProperty(Object.constructor.prototype, 'name', {
      get: function () {
        return this.toString().trim().replace(/^\S+\s+(\w+)[\S\s]+$/, '$1')
      }
    })
  }
```

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/f3a6d5ce75bb9d60b87074240838f5429e896b60
